### ISSUE TITLE
Make Helm chart name lower case for usages in resource names

### ISSUE THREAD

**Issue by rinormaloku** [open-telemetry/opentelemetry-helm-charts#656]
When using OTel collector as a sub-chart you can use some more descriptive name, such as:

values.yaml file
```
  # subchart values
  telemetryCollector:
  ...
  telemetryGateway:
  ...
```

This will produce invalid output. Because the chart-name, is used here:
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/af9d9c74bae6f2042d4687d09f1b92d3af23f104/charts/opentelemetry-collector/templates/_pod.tpl#L10

And that field needs to follow a DNS label standard: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names

This would be fixed by simply lower casing all usages of Chart-Name.

**Comment by TylerHelmuth** [comment]
Instead of changing the chart is it possible to use subchart names that are valid container names like telemetrycollector or telemetry-collector?

**Comment by rinormaloku** [comment]
@TylerHelmuth yes it is possible, we are doing that. So this is in no way urgent.

That said, two points why I'd recommend that we merge it:

1. Let's stick with the convention set by Kubernetes: 
- Kubernetes camelCase for naming properties. (That's not a convention for YAML, but one for YAML in the Kubernetes API)
- Helm is a tool used for templating Kubernetes resources.
- Hence, UX would be better if camelCase is used. (btw this will be an issue for any chart that uses this as a dependency)

2. This change is backward compatible.
- _Either way_ lower-case had to be used and is used
- We are only resolving issues for people who are (or will be) using it as a sub-chart, with full backward compatibility


Meaning that there are some benefits in UX for merging this PR and there are no downsides.

**Comment by TylerHelmuth** [comment]
@rinormaloku that is a compelling argument, thank you for sharing.  Do you have examples of how other charts handle this?  If use we lower Chart.Name some places it would probably be best to be consistent and use it everywhere, do you agree?

**Comment by rinormaloku** [comment]
In the case of the container name, most of the charts apply a static value, because the name of the container doesn't matter, and it's actually easier to navigate through different otel collector deployments if the names of the containers are the same. Meanwhile, the different deployment names differ, which is what we want.

Similarly, this volume name, that's used only as a reference can be static. e.g. otel-configuraiton (natch that the configmap itself has a different name)
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/0bc331cf56b97d574ca6a2a23a41211f1ff65f72/charts/opentelemetry-collector/templates/_pod.tpl#L134

Though, these changes would mean that it will trigger a deployment rollout.

---

I assume that one requirement for us is **to be fully backward compatible and not trigger a rollout.** If that's the case there is no way around of just making it lowercase and not making further modifications.

I psuhed another commit. In it the `chart.name | lower` section is moved into a template, and included in all the needed locations.

**Comment by TylerHelmuth** [comment]
closed by https://github.com/open-telemetry/opentelemetry-helm-charts/pull/657
